MeesGame/Gameplay/GameObjects/TilesTypes/DoorTile.cs

```diff
@@ -51,26 +70,32 @@ public override bool IsActionForbiddenFromHere(Player player, PlayerAction actio
         public void OpenDoor(Player player)
         {
             //A player is only allowed to open a door if he has the right key.
-            if (player.HasKey())
+            if (player.HasItem(doorColor.ToInventeryItemType()))
             {
                 doorIsOpen = true;
             }
         }
 
         public override void UpdateGraphicsToMatchSurroundings()
         {
+            int sheetIndex = 0;
```

Comment:
It's not a sheet index... it may be better to count the number of horizontally connected walls minus the number of vertically connected walls. That should also make the code a bit simpler.